### ISSUE TITLE
4C4107F3 - LEGO Star Wars III - The Clone Wars

### ISSUE THREAD

**Issue by anonymousasimov** [xenia-canary/game-compatibility#236]
### Validation

- [X] I've read the [FAQ](https://github.com/xenia-canary/xenia-canary/wiki/FAQ).
- [X] The Xenia build used is from a canary branch. (not MLBS/AlexVS/master/pull requests, etc)
- [X] This issue isn't for help (tech support) with Xenia, or a game.
- [X] I've checked if this game doesn't already have a compatibility report.

### Xenia version

https://github.com/xenia-canary/xenia-canary/commit0ad1e3d5855cdf8e3c4ee20f7d3ba5ebc20c2edd

### Issues

Hangs consistently on scene in cutscene of first level - as shown in screenshot. Does not crash.

### Log

[xenia.zip](https://github.com/user-attachments/files/16691592/xenia.zip)

### Screenshots

![image](https://github.com/user-attachments/assets/8f1ea17b-2230-47e6-aff4-c87a2658df8d)

### Labels

state-hang

**Comment by metlobster** [comment]
same prob... did you ever fix it???

**Comment by legoj15** [comment]
Bypassing the crash with a save from a physical console results in an inevitable polygon explosion that results in the game being unplayable anyways. Enabling "Clear GPU Cache" does not resolve the issue. Tested on https://github.com/xenia-canary/xenia-canary/commit/de3b49e356ffdafa365afad49f3ed4a6959a5990

![Image](https://github.com/user-attachments/assets/384a28e3-9e33-4125-b918-cb9fd1ecdbc6)

[xenia.log](https://github.com/user-attachments/files/20140659/xenia.log)

**Comment by Sowa95** [comment]
https://github.com/xenia-canary/xenia-canary/commit/de3b49e356ffdafa365afad49f3ed4a6959a5990

[xenia.zip](https://github.com/user-attachments/files/20141190/xenia.zip)

- softlocks in longer cutscenes due to audio, also music dies in menus and gameplay.
- requires `readback_memexport` to prevent corrupted graphics.
- game speed tied to FPS.

Played about 45 minutes. Briefly visited 3 starting planets and was about to finish Duel of the Droids, but cutscene froze, so I quit.

<details><summary>Readback changes (outdated)</summary>
<p>

This game used to require readback_resolve.

[8569569](https://github.com/xenia-canary/xenia-canary/releases/tag/8569569) or older - readback_memexport
[b9be601](https://github.com/xenia-canary/xenia-canary/releases/tag/b9be601) to [70e44ab](https://github.com/xenia-canary/xenia-canary-releases/releases/tag/70e44ab) - readback_resolve
[e2c3368](https://github.com/xenia-canary/xenia-canary-releases/releases/tag/e2c3368) or newer - readback_memexport

</p>
</details>

### 16580f5 audio regression

tldr:
[c9cd6f1](https://github.com/xenia-canary/xenia-canary/commit/c9cd6f15fc251b43ef461c993be7d11ac7f18788) - no issues with audio
[16580f5](https://github.com/xenia-canary/xenia-canary/commit/16580f5fae0b13e982708023a239033443c3f5d2) - audio dead. It recovered a bit later on, but not completely. New XMA decoder doesn't really change anything, it's better in menus, but worse for cutscenes.

<details><summary>Details</summary>
<p>

Everything is repeatable. Audio dead always at the same time depending on the build/decoder.

1st cutscene (excluding skippable part):
c9cd6f1 - plays whole cutscene and loads to gameplay.
from 16580f5 to 97fdf9c - freeze in 19s.
b2dd489 - improved, freeze in 59s
b2dd489 is the current state of latest Canary. New XMA decoder is worse, freeze in 32s, cutscenes for the first 3 planets are also worse. Especially Battle of Genosis, which plays fully on old but freeze on new.

Music in menu:
c9cd6f1 - no issues within 10 minutes, probably plays endlessly.
from 16580f5 to 97fdf9c - music dead in 2 s.
b2dd489 - improved, music dead in 4 s.
b2dd489 is the current state of latest Canary. New XMA decoder is much better in menu, music plays for 29s.

Music in gameplay, main hub:
c9cd6f1 - no issues within 10 minutes, probably plays endlessly.
from 16580f5 to 97fdf9c - music dead in 30s.
b2dd489 - improved, music dead in 54s.
b2dd489 is the current state of latest Canary. Depending on the track, new XMA decoder could be slightly worse or slightly better. For example in main hub it plays ~3s shorter, but in the Battle for Genosis ~3s longer.

</p>
</details>

### Screenshots

![Image](https://github.com/user-attachments/assets/2b9661d2-e3c9-4ae8-ba8a-a6f661a0ba81)

<details><summary>More screenshots</summary>
<p>

![Image](https://github.com/user-attachments/assets/1f17ddb5-66f0-487b-a1c5-f696d4ace117)
![Image](https://github.com/user-attachments/assets/f4fc3f18-88a6-4fb2-9287-00a3d292ed20)
![Image](https://github.com/user-attachments/assets/5dbcfaac-898f-46eb-9ba2-e795c43c305e)
![Image](https://github.com/user-attachments/assets/39775b81-3690-4e36-90f3-841d3f0c2c41)

</p>
</details>

apu-garbage | gpu-drawing-corrupt | gpu-readback | regression | vsync-off-speedup

**Comment by BorealisEcho** [comment]
Tested on 99ea6da_canary_experimental,

Still has softlocking/hanging issues in the first cutscene regardless of xma_decoder and use_dedicated_xma_thread.

[xenia.zip](https://github.com/user-attachments/files/28098846/xenia.zip)

<img width="1280" height="720" alt="Image" src="https://github.com/user-attachments/assets/82f5d85b-bece-4650-9efe-8f1d87487234" />
